server/src/controllers/course.ts

```diff
@@ -40,3 +53,146 @@ export async function addCourse(req: Request, res: Response): Promise<void> {
     });
   }
 }
+
+export async function getCourse(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: number = Number(req.params.courseId);
+    const course: courseService.CourseWithTranslationAndInstance = await courseService.findCourseById(courseId);
+  
+    const courseData: CourseData = {
+      id: course.id,
+      courseCode: course.courseCode,
+      minCredits: course.minCredits,
+      maxCredits: course.maxCredits,
+      department: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      name: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      evaluationInformation: {
+        en: '',
+        fi: '',
+        sv: ''
+      }
+    };
+  
+    course.CourseTranslations.forEach((translation: CourseTranslation) => {
+      switch (translation.language) {
+      case Language.English:
+        courseData.department.en = translation.department;
+        courseData.name.en = translation.courseName;
+        break;
+      case Language.Finnish:
+        courseData.department.fi = translation.department;
+        courseData.name.fi = translation.courseName;
+        break;
+      case Language.Swedish:
+        courseData.department.sv = translation.department;
+        courseData.name.sv = translation.courseName;
+        break;
+      }
+    });
+  
+    res.status(200);
+    return res.send({
+      success: true,
+      course: courseData
+    });
+  } catch (error: unknown) {
+    console.log(error);
+  
+    if (error instanceof Error) {
```

Comment:
Could we have more fine-grained checking of the kind of Error here? I think a database error, for instance, shouldn't cause a 404, and that's not guaranteed very strongly here.

Reply:
updated errors https://github.com/aalto-grades/base-repository/pull/112/commits/a3fa4c871221cae93c332da6d710e7631ff39df4